aces/Loris
# [examiner] Adds permission check when adding examiner to site.

## Brief summary of changes
Permissions are check now before allowing to add examiner to site. 
Before this PR, a user with only the permission `Examiner: Add and Certify Examiners - Own Sites` was able to add examiners to sites it don't belongs to. Now this feature is only granted to users with the corresponding level of permission.

#### Testing instructions (if applicable)

1. With a user that only have the permission `Examiner: Add and Certify Examiners - Own Sites`  try to create a new examiner in `MainMenu->Clinical->Examinier`
2.  When clicking the Add Examiner button the modal windows that opened should only list the sites the current user belongs to.
3.  See image as reference.
![image](https://github.com/aces/Loris/assets/37309344/e62970b9-e160-4fdf-bb1b-723f5e709f17)
4. Only for the listed sites the current user should be able to add an examiner.
#### Link(s) to related issue(s)

* Resolves #9149
